domain/resource/state/resource.go

```diff
@@ -388,17 +397,21 @@ WHERE ur.resource_uuid = $resourceIdentity.uuid`
 		// Process each resource from the application to check polled state
 		// and if they are associated with a unit.
 		for _, res := range resources {
-			resId := resourceIdentity{UUID: res.UUID, ApplicationUUID: res.ApplicationUUID}
 
-			// Check to see if the resource has already been polled.
-			err = tx.Query(ctx, checkPolledStmt, resId).Get(&resId)
-			if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
-				return errors.Capture(err)
+			if res.State == statePotential {
```

Comment:
In my WIP resources migration PR I've added `StatePotential` and `StateAvailable` as a constants in `domain/resource/type.go`, might as well just do that here, that way they are also deserving of longer comments explaining their purpose.